components.md

```diff
@@ -0,0 +1,79 @@
+# Component Usage
+
+### Overview
+
+This document describes the concepts required for creating Components to run on the BOS Web Engine.
+
+### Terms
+- **Outer Application** is the application responsible for rendering Components and brokers messages between containers.
+- **Components** refer to the functions returning JSX to be rendered in BOS Web Engine.
+- **Root Component** refers to the top-level Component loaded by the outer application.
+- **Containers** provide the runtime context for one or more Components. They are implemented as sandboxed `<iframe>`s,
+communicating with other containers via the outer application's `window.postMessage` method.
+- **Component ID**s uniquely identify Components rendered within a parent Component. They are roughly analogous to `key`
+in React.
+
+### Loading Mode
+
+There are two modes for loading a Component: **sandboxed** and **trusted**. These modes make it possible to create boundaries
+within Component trees, granting developers more control over the balance of performance and security in their applications.
+
+#### Sandboxed
+
+Sandboxed Components are run in a dedicated container, independent of their parent Component's context. All communication
+between parent and child (e.g. re-rendering, `props` method invocation) is handled via `postMessage` communication through
+the outer application.
+
+#### Trusted
+
+When a Component is loaded as **trusted**, the parent Component loads the child Component and its descendants within its
+own container. The definitions of all Components in the child Component tree are inlined into the parent Component definition.
+This approach avoids the overhead in rendering via event propagation at the cost of executing external JSX code within the
+same context.
+
+#### Usage
+
+By default, Components are loaded in **sandboxed** mode. To opt in to **trusted** loading, use the `isTrusted` prop:
+```jsx
+<Widget isTrusted={true} />
```

Comment:
nit: `<Widget isTrusted />` would also work, right? If so, I think that shorthand is nicer